frontend/src/pages/admin/questions/update/[id].tsx

```diff
@@ -0,0 +1,258 @@
+              <h1 className="form-title">Update Question</h1>
+              <div className="mt-3">
+                <label className="form-label">Category *</label>
+                <select 
+                  className="form-control" 
+                  tabIndex={1}
+                  disabled={isSubmitting}
+                  {...register('category_id', { required: 'Category is required' })}
+                >
+                  <option value={category?.id}>{category?.title}</option>
+                </select>
+                {errors?.category_id && <span className="error">{`${errors?.category_id?.message}`}</span>}
+              </div>
+              <div className="mt-6">
+                <label className="form-label">Question *</label>
+                <input 
+                  type="text" 
+                  className="form-control" 
+                  placeholder="Write your question"
+                  tabIndex={2}
+                  disabled={isSubmitting}
+                  defaultValue={question?.value}
+                  {...register('value', { required: 'Question is required' })}
+                />
+                {errors?.value && <span className="error">{`${errors?.value?.message}`}</span>}
+              </div>
+              <div className="mt-3">
+                <label className="form-label">Answer *</label>
+                <select 
+                  className="form-control" 
+                  tabIndex={1}
+                  disabled={isSubmitting}
+                  {...register('choice_id', { required: 'Right Answer is required'})}
+                >
+                {choices?.map(({ id, value }: IChoice, i: number) => 
+                  <option key={id} value={i+1} selected={question?.choice_id == i+1}>{value}</option>
+                )}
+                </select>
+                {errors?.choice_id && <span className="error">{`${errors?.choice_id?.message}`}</span>}
+              </div>
+              <div className="mt-6">
+                <label htmlFor="message" className="form-label">Choices *</label>
+                <ChooseFields
+                  choices={choices}
+                  actions={{ 
+                    handleAddField,
+                    handleChangeInput, 
+                    handleRemoveField, 
+                  }}
+                  isSubmitting={isSubmitting}
+                />
+              </div>
+              <div className="mt-4 flex justify-end">
+                <button 
+                  type="submit" 
+                  tabIndex={3}
+                  disabled={isSubmitting}
+                  className="btn-success px-10 py-3"
+                >
+                  {isSubmitting ? (
+                    <Spinner className="w-5 h-5" />
+                  ) : 'Save'}
+                </button>
+              </div>
+            </form>
+          )}
+        </section>
+      </main>
+    </Layout>
+  )
+}
+
+type ChooseProps = {
```

Comment:
can you refactor and move the ChooseFields to another file? move it in a components folder in the same folder as the [id].tsx